src/lib.rs

```diff
@@ -1 +1,40 @@
+//! # Plonk Gadgets
+//! This library cointains the gadgets that the Dusk-Network protocol needs to build it's ZK-Circuits.
+//! The library **contains generic gadgets** which are used across Dusk's tech stack, all of the other
+//! gadgets used which depend on foreign types are placed on the libraries where this types are defined.
+//!
+//!
+//! ## WARNING
+//! This implementation is not audited. Use under your own responsability.
+//!
+//! ## Content
+//! This library provides:
+//!
+//! - Scalar gadgets: `is_non-zero`, `maybe_equals`, `conditionally_select_one`, `conditionally_select_zero`.
+//! - Range gadgets: `range_check`, `max_bound`.
+
+#![doc(
+    html_logo_url = "https://lh3.googleusercontent.com/SmwswGxtgIANTbDrCOn5EKcRBnVdHjmYsHYxLq2HZNXWCQ9-fZyaea-bNgdX9eR0XGSqiMFi=w128-h128-e365"
+)]
+#![doc(html_favicon_url = "https://dusk.network/lib/img/favicon-16x16.png")]
+#![allow(clippy::suspicious_arithmetic_impl)]
+// Some structs do not have AddAssign or MulAssign impl.
+#![allow(clippy::suspicious_op_assign_impl)]
```

Comment:
I think we don't have any code in this repo that needs them, double check and if that is the case please remove them.

Reply:
I left the `suspicious_arithmetic_impl` since we have bitshift ops.